sway/commands/bind.c

```diff
@@ -196,12 +196,122 @@ static struct cmd_results *identify_key(const char* name, bool first_key,
 	return NULL;
 }
 
+static struct cmd_results *switch_binding_add_to_list(struct sway_switch_binding *binding,
+		const char *bindtype, const char *switchcombo, bool warn) {
+
+	list_t *mode_bindings = config->current_mode->switch_bindings;
+
+	// overwrite the binding if it already exists
+	bool overwritten = false;
+	for (int i = 0; i < mode_bindings->length; ++i) {
+		struct sway_switch_binding *config_binding = mode_bindings->items[i];
+		if (binding_switch_compare(binding, config_binding)) {
+			sway_log(SWAY_INFO, "Overwriting binding '%s' to `%s` from `%s`",
+					switchcombo, binding->command, config_binding->command);
+			if (warn) {
+				config_add_swaynag_warning("Overwriting binding"
+						"'%s' to `%s` from `%s`",
+						switchcombo, binding->command,
+						config_binding->command);
+			}
+			free_switch_binding(config_binding);
+			mode_bindings->items[i] = binding;
+			overwritten = true;
+		}
+	}
+
+	if (!overwritten) {
+		list_add(mode_bindings, binding);
+	}
+
+	sway_log(SWAY_DEBUG, "%s - Bound %s to command `%s`",
```

Comment:
I think this should be in the `!overwritten` body. The `Overwriting binding` log message should be sufficient.